File: java-samples/Features/cordaservice-autopayroll/contracts/src/test/java/net/corda/samples/autopayroll/contracts/StateTests.java
package net.corda.samples.autopayroll.contracts;
import net.corda.samples.autopayroll.states.MoneyState;
import net.corda.testing.node.MockServices;
import org.junit.Test;

public class StateTests {
    private final MockServices ledgerServices = new MockServices();

    @Test
    public void hasFieldOfCorrectType() throws NoSuchFieldException {
        // Does the message field exist?
        MoneyState.class.getDeclaredField("amount");
        // Is the message field of the correct type?
        assert(MoneyState.class.getDeclaredField("amount").getType().equals(int.class));
    }
}

File: kotlin-samples/Features/cordaService-autopayroll/contracts/src/test/kotlin/net/corda/samples/autopayroll/StateTests.kt
package net.corda.samples.autopayroll

import net.corda.samples.autopayroll.states.MoneyState
import net.corda.testing.node.MockServices
import org.junit.Assert
import org.junit.Test

class StateTests {
    private val ledgerServices = MockServices()

    @Test
    fun hasFieldOfCorrectType() {
        // Does the message field exist?
        MoneyState::class.java.getDeclaredField("amount")
        // Is the message field of the correct type?
        Assert.assertEquals(MoneyState::class.java.getDeclaredField("amount").type, Int::class.java)
    }
}